proc_scripts/load_data.py

```diff
@@ -52,27 +90,137 @@ def proc_spincore_CPMG_v1(s, fl=None):
+    p90_s = s.get_prop('acq')['P'][1]*1e-6
+    quad_pts = ndshape(s)['t2'] # note that we have not yet chunked t2
+    nPoints = quad_pts/nEchoes
+    acq_time = dwdel2*nPoints*2
+    # {{{ these are hard-coded for the pulse sequence
+    #     if we need to, we could pull these from the pulse sequence, as we do
+    #     for anavpt above
+    tau_extra = 20e-6
+    tau_pad_start = tau_extra-dwdel1-6e-6
+    tau_pad_end = tau_extra-6e-6
+    twice_tau = 2*p90_s + 5e-6 + tau_pad_start + 1e-6 + acq_time + tau_pad_end +1e-6
+    #twice_tau should be the period from one 180 to another
+    # }}}
+    s.set_units('t2','us')
+    s.chunk('t2',['tE','t2'],[nEchoes,-1])
+    s.setaxis('tE', (1+r_[0:nEchoes])*twice_tau)
+    s.ft('t2', shift=True).ft(['ph1','ph2'])
+    s.reorder(['ph1','ph2','indirect'])
+    if fl is not None:
+        fl.next('freq domain coh domain')
+        fl.image(s)
+    s.ift('t2')
+    if fl is not None:
+        fl.next('t2 chunked', figsize=(5,20))
+        fl.image(s)
+    s.ft('t2')
+    return s
+
+def proc_bruker_CPMG_v1(s,d12,tau_extra,fl=None):
+    logger.info(strm("pre-processing CPMG acquired on Bruker"))
+    s.chunk('indirect',['ph1','ph2','indirect'],[4,2,-1])
+    s.setaxis('ph1',r_[0,4]/4.)
+    s.setaxis('ph2',r_[0,2]/2.)
+    if fl is not None:
+        fl.next('raw data before')
+        fl.image(s)
+    s.ft(['ph1','ph2'])
+    s.reorder(['ph1','ph2','indirect','t2'])
+    anavpt_info = [j for j in s.get_prop('pulprog').split('\n') if 'anavpt' in j.lower()]
+    anavpt_re = re.compile(r'.*\banavpt *= *([0-9]+)')
+    anavpt_matches = (anavpt_re.match(j) for j in anavpt_info)
+    for m in anavpt_matches:
+        if m is not None:
+            anavpt = int(m.groups()[0])
+    actual_SW = 20e6/anavpt # JF: check that this is based on the manual's definition of anavpt
+    bruker_final_t2_value = double(s.getaxis('t2')[-1].item())
+    s.setaxis('t2',1./actual_SW*r_[0:ndshape(s)['t2']]) # reset t2 axis to true values based on anavpt
+    logger.debug(strm("the final t2 value according to the Bruker SW_h was",
+        bruker_final_t2_value, "but I determine it to be",
+        double(s.getaxis('t2')[-1].item()), "with anavpt"))
+    nEchoes = s.get_prop('acq')['L'][25]
+    dwdel1 = s.get_prop('acq')['DE']*1e-6
+    dwdel2 = (anavpt*0.05e-6)/2
+    #d12 is read as 0 if taken from parameters bc its too small
+    d12 = d12     
+    d11 = s.get_prop('acq')['D'][11]
+    p90_s = s.get_prop('acq')['P'][1]*1e-6
+    quad_pts = ndshape(s)['t2'] # note that we have not yet chunked t2
+    nPoints = quad_pts/nEchoes
+    acq_time = dwdel2*nPoints*2
+    # {{{ these are hard-coded for the pulse sequence
+    #     if we need to, we could pull these from the pulse sequence, as we do
+    #     for anavpt above
+    tau_extra = tau_extra
+    tau_pad_start = tau_extra-dwdel1-6e-6
+    tau_pad_end = tau_extra-6e-6
+    twice_tau = 2*p90_s + 5e-6 + tau_pad_start + 1e-6 + acq_time + tau_pad_end +1e-6
```

Comment:
so we would not know if this number was wrong, so we want to double-check it.  You should also be able to calculate this from the acquisition length and the padding on either side of  the acquisition.  We should double-check that both are equal with an `assert` statement.  You might need to get @aabeaton 's help to interpret what I'm saying here + how to implement it.

Reply:
I pulled these variables straight from the pulse program. These are how they are defined there. At least up until twice_tau. 
![image](https://user-images.githubusercontent.com/52217688/104939065-f15b2b00-597d-11eb-9ff4-09cfd62e54da.png)
this is the pulse sequence timing where we calculate twice tau

Reply:
this doesn't address my point, which is that twice tau should also be equal to
`dwdel1+5e-6+tau_pad_start+1e-6+dwdel2*num_pts+tau_pad_end+1e-6+5e-6` (the time from one 180 pulse to the next) --> so we want an assertion statement that also checks that it's equal to this expression

Reply:
I'm confused... we are asserting that twice_tau=twice_tau?

Reply:
your expression involves the 90 time, while mine does not  -- they are not the same expression

Reply:
more specifically -- tau should be equal to two time periods:

- the equivalent evolution time during the 90 pulse (given by Cavanagh -- 2*t90/π, from pg 266 Eq. [3.199]), plus the time from the end of the ninety up to the start of the first 180 (what I think you calculate above)
- half of the total time from the end of one 180 pulse until the start of the next

Reply:
okay so since I am using twice_tau this statement should follow those bullet points correct?

Reply:
or did I forget to divide 2*p_90 by pi?

Reply:
yes, it should follow those statements & it does look like you forgot to divide by pi

Reply:
actually, why not just define two variables -- `ninety_through_180` and `between_180s`, and then check `assert ninety_through_180 == between_180s`

(`twice_tau` is ambiguous in the current discussion)